pdsClient/pdsClient.go

```diff
@@ -192,6 +236,76 @@ func (c *E3dbPDSClient) InternalGetRecord(ctx context.Context, recordID string)
 	return result, err
 }
 
+// CreateSharingAccessKey attempts to create an access key for the specified reader to be able to decrypt records of the specified type, returning error (if any).
+func (c *E3dbPDSClient) CreateSharingAccessKey(ctx context.Context, params CreateSharingAccessKeyRequest) error {
+	// Get the current encrypted access key
+	// used to write records of this type by
+	// the client specified in params
+	getAccessKeyResponse, err := c.GetAccessKey(ctx, GetAccessKeyRequest{
+		WriterID:   params.WriterID,
+		UserID:     params.UserID,
+		ReaderID:   params.UserID,
+		RecordType: params.RecordType,
+	})
+	if err != nil {
+		return err
+	}
+	encryptedAccessKey := getAccessKeyResponse.EAK
+	if encryptedAccessKey == "" {
+		return errors.New(fmt.Sprintf("no access key exists for records of type %s", params.RecordType))
```

Comment:
Is this method a helper method for tests?

In the python SDK, if no EAK exists it is created and put for both the sharer and sharee to allow sharing of record types that don't yet exist

Reply:
Yes, and tests have a MakeClientWriterForRecord that creates an access key for the sharer/writer of a record.